consensus/replay.go

```diff
@@ -132,9 +136,13 @@ func (cs *ConsensusState) catchupReplay(csHeight int64) error {
 		msg, err = dec.Decode()
 		if err == io.EOF {
 			break
+		} else if IsDataCorruptionError(err) {
+			cs.Logger.Debug("data has been corrupted in last height of consensus WAL", "err", err, "height", csHeight)
+			cmn.PanicCrisis(fmt.Sprintf("data has been corrupted (%v) in last height %d of consensus WAL", err, csHeight))
```

Comment:
are we sure we want to continue operation here?

Reply:
maybe not. but then whats a user to do? inspect the wal and delete what they think are the troublesome bytes? how can they even do that since it's binary? they would need to `wal2json` and then `json2wal` ... do we have the latter?

Reply:
this returns an error if the wal is empty. so we'd have to not. 

Maybe we just do what kyle suggested - we tell the user it failed and to run `tendermint repair wal` or similar, which just removes the broken line so they can try to run it again.

Reply:
Ok. We should just point them to the corruption docs :)

Reply:
PanicCrisis etc are deprecated, lets just panic.

Reply:
If every line were `<hex(sha256(msg-bytes)[:4])>:<hex(msg-bytes)>`
e.g. `00112233:0102030405060708...`
where 00112233 is the sha256(msg-bytes)[:4] checksum,
we can easily check sha256(line[4:]) and see if it matches the checksum.
would this solve the corruption issue?